clang/test/CodeGenSYCL/simple-sycl-virtual-function.cpp

```diff
@@ -0,0 +1,30 @@
+// This test checks that the FE generates global variables corresponding to the
+// virtual table in the global address space (addrspace(1)) when
+// -fsycl-allow-virtual-functions is passed.
+
+// RUN: %clang_cc1 -triple spir64 -fsycl-allow-virtual-functions -fsycl-is-device -emit-llvm %s -o - | FileCheck %s
```

Comment:
```suggestion
// RUN: %clang_cc1 -triple spir64 -fsycl-allow-virtual-functions -fsycl-is-device -emit-llvm %s -o - | FileCheck %s  --check-prefixes CHECK,CHECK-PTR
// RUN: %clang_cc1 -triple spir64 -fsycl-allow-virtual-functions -fsycl-is-device -fexperimental-relative-c++-abi-vtables -emit-llvm %s -o - | FileCheck %s --check-prefixes CHECK,CHECK-REL
```

to cover the latest change, the 2 last checks will be different (hence the check-ptr, check-rel)

Reply:
First of all - I've tested this thing and as I understand it, in this case (my test) there is no need for `bitcast`, so we only see the generation of another "relative" GEP instead of "ptr" GEP. Is it OK for us? If so, I'll push this commit.
The second thing - I tried to compile my E2E test with this key (`-fexperimental-relative-c++-abi-vtables`) and there is some failure in llvm-spirv, so anyway we need to adjust FE (or translator) to support this option - and this work is not for this patch.

Reply:
> I've tested this thing and as I understand it, in this case (my test) there is no need for bitcast, so we only see the generation of another "relative" GEP instead of "ptr" GEP. Is it OK for us? 

make sense if the bitcast does nothing it will not emit the node. So that's indirectly testing it, so I think so.

> I tried to compile my E2E test with this key (-fexperimental-relative-c++-abi-vtables) and there is some failure in llvm-spirv, so anyway we need to adjust FE (or translator) to support this option - and this work is not for this patch.

I agree it shouldn't be part of the patch. My point is only about testing the paths changed (so checking you don't insert a bitcast that changes the address space of a pointer).

Reply:
Okay, done in [d04efa2](https://github.com/intel/llvm/pull/7255/commits/d04efa2c481da2438096d5a7aabc3376b4dcb911)